### ISSUE TITLE
Machine Learning in theory and in practice

### ISSUE THREAD

**Issue by SanyHe** [ZJUEarthData/geochemistrypy#45]
the reports or related summaries to be documented by members!

**Comment by Qinyh1** [comment]
[Notes of Chapter1-2 for “Hands-on Machine Learning with Scikit-Learn, Keras, and TensorFlow”.docx](https://github.com/ZJUEarthData/geochemistrypy/files/9052313/Notes.of.Chapter1-2.for.Hands-on.Machine.Learning.with.Scikit-Learn.Keras.and.TensorFlow.docx)

In _Hands-on Machine Learning with Scikit-Learn, Keras, and TensorFlow_, chapter 1 mainly introduces the concept, classification, and challenges of machine learning, while chapter 2 shows how to carry on a whole end to end project (i.e., building a model for predicting housing price of the California). We can have a general understanding about the machine learnig through this note.

**Comment by Kaixin7** [comment]
Chapter 2 introduces what a Machine Learning project looks like and how to use tools to train a model, which helps me carry out a practical application on Fe3+ project. There are train data and test data on the table. And I use Linear Regression model to train elements data and predict  Fe3+/Fetot value. Firstly I got an R2 value of 0.8649. But I found that  R2 decreased after I normalized train data using standardization method. So I tried other methods and the L1 norm successful improved R2 from 0.865 to 0.895. The codes to train the model are in the attachment.
To further improve the value of R2, one approach is to create new characteristic of train data, which needs more professional geochemistry knowledge. The other one is that, the number of train data can be increased to get a greater model, so that we can use it to obtain better predictions
[Fe.py.zip](https://github.com/ZJUEarthData/geochemistrypy/files/9399805/Fe.py.zip)
.
